text/0040-token-metadata.md

```diff
@@ -0,0 +1,167 @@
+which can be shipped with clients, as a stop-gap measure.)
+
+If the clients can dynamically query this information, then a client update is not required for clients to make use of new tokens which are
+created, which significantly reduces the amount of time it takes for a new token to be available to users.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Token metadata includes the following:
+
+* Short (exchange ticker) name of a token, e.g. MOB. (This is not expected to be translated.)
+* Long (english) name of a token, e.g. MobileCoin. (This is expected to be translated, but the client must maintain the translation list.)
+* Display conversion factor, e.g. 10^12. This conversion factor converts from the smallest representable units (`u64`) which appear in the blockchain,
+  to the user-displayable unit. (Note that this is possibly not a power of 10.) `u64` amounts
+  which are obtained from `TxOut`'s on chain should be divided by this number, for the token id, before being displayed to the user.
+* (Optional) URL pointing to an SVG logo of the token. (This is expected to be scalable and appropriate for display on devices of many form factors.)
+* (Optional) URL pointing to an MCIP which proposed the token.
+
+The consensus network will provide an endpoint at which this metadata can be requested for one or more token ids.
+
+This data will be provided at startup time via a file to the consensus node. (In the future, we may consider to store this data on chain.)
+The node should require this data for any token which is configured to exist (has a minimum fee in `tokens.toml`).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+A token metadata proto has these fields
+
+```
+message TokenMetadata {
+    // The short name is a standard ticker symbol for the token
+    String short_name = 1;
+    // The long name is the standard english name for the token
+    String long_name = 2;
+    // The conversion factor is an unsigned 64 bit number expressed as a decimal string
+    String display_conversion_factor = 3;
+    // If present, a url pointing to a logo for the token
+    String logo_url = 4;
+    // If present, a url pointing to an MCIP which proposed the token
+    String mcip_url = 5;
+}
+```
+
+A consensus network client-facing API will be exposed to query for these objects given one or more token ids.
+
+The `tokens.toml` file schema is extended to contain these fields per token. The minting trust root should sign
+these fields.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This proposal
+
+* Adds to the complexity of configuring a network
+* Does not specify a mechanism that forces different nodes to have matching values
+  (we could choose to hash this data into responder id for instance)
+* Does not commit this data to the blockchain
+* Does not prevent the metadata from being changed -- only trust in the foundation not to sign another value
```

Comment:
the node operators would still need to accept the new file so the power is not fully in the foundation's hands